frontend/src/app/projects/project-detail/model-overview/model-detail/git-model.service.ts

```diff
@@ -16,8 +16,10 @@ export class GitModelService {
 
   constructor(private http: HttpClient) {}
 
-  private _gitModel = new Subject<GetGitModel>();
-  public _gitModels = new BehaviorSubject<Array<GetGitModel>>([]);
+  private _gitModel = new Subject<GetGitModel | undefined>();
+  public _gitModels = new BehaviorSubject<Array<GetGitModel> | undefined>(
+    undefined
+  );
```

Comment:
```suggestion
  public _gitModels = new BehaviorSubject<GetGitModel[] | undefined>(undefined);
```
First I would remove the `Array<...>` since  we usually use the `[]` notation.

Also, I'm a little confused that the `_gitModels` behavior subject is public, since the main idea here is to use the `gitModels` observable outside of the service. There is only one place where we use `_gitModels` directly, but the initial idea was to use a local `gitModels`: GetGitModel[] = undefined` in components and set it by subscribing to the public `gitModels` observable and using `loadGitModels()`.

Reply:
> but the initial idea was to use a local `gitModels: GetGitModel[] = undefined` in components and set it by subscribing to the public `gitModels` observable and using `loadGitModels()`.

When we just want to get the value of `_gitModels`, I dislike the idea of replicating the whole `subscribe`, `unsubscribe` and a separate variable (which has the same value of the `gitModel` anyway then. But as we have no consistent decision on that, I'd skip it for now.